source/common/common/hash.h

```diff
@@ -26,6 +27,42 @@ class HashUtil {
     return XXH64(input.data(), input.size(), seed);
   }
 
+  /**
+   * Return 64-bit hash from deterministically serializing a value in
+   * an endian-independent way and hashing it with the xxHash algorithm.
+   *
+   * Floats and doubles are just done as endian-adjusted raw bytes the same
+   * as integers; this is on par with what protobuf serialization does.
+   * src/google/protobuf/generated_message_util.cc literally just treats
+   * a float the same as int32 for serialization purposes.
+   *
+   * enums are excluded because they're not needed; if they were to be
+   * supported later they should have a separate function because enum
+   * sizes are implementation-specific.
+   *
+   * bools have an inlined specialization below because they too have
+   * implementation-specific sizes.
+   *
+   * @param input supplies the value to hash.
+   * @param seed supplies the hash seed which defaults to 0.
+   * See https://github.com/Cyan4973/xxHash for details.
+   */
+  template <
+      typename ValueType,
+      std::enable_if_t<std::is_scalar_v<ValueType> && !std::is_enum_v<ValueType>, bool> = true>
```

Comment:
I assume this means that if a string or nested message is passed in, this function will not compile. Is that right?